budgetportal/import_export_admin.py

```diff
@@ -0,0 +1,148 @@
+from budgetportal.models import Department, Government, Sphere
+from django import forms
+from django.core.exceptions import ValidationError
+from django.utils.text import slugify
+from import_export import resources
+from import_export.admin import ImportForm
+from import_export.fields import Field
+from import_export.formats import base_formats
+from import_export.instance_loaders import ModelInstanceLoader
+from import_export.widgets import Widget
+
+
+class CustomIsVotePrimaryWidget(Widget):
+    """
+    Widget for converting is_vote_primary fields.
+    """
+
+    def render(self, value, obj=None):
+        if value:
+            return "True"
+        else:
+            return "False"
+
+    def clean(self, value, row=None, *args, **kwargs):
+        if value is None or value == "" or value.upper() == 'TRUE':
+            return True
+        else:
+            return False
+
+
+class CustomGovernmentWidget(Widget):
+    """
+    Widget for converting government_name and sphere fields to governments.
+    """
+
+    def set_sphere(self, sphere):
+        try:
+            self.sphere = Sphere.objects.get(id=sphere)
+        except Sphere.DoesNotExist:
+            raise ValidationError('Sphere with id %s does not exist.' % sphere)
+
+    def render(self, value, obj=None):
+        return value.name
+
+    def clean(self, value, row=None, *args, **kwargs):
+        try:
+            government = Government.objects.get(
+                sphere=self.sphere,
+                slug=slugify(value),
+            )
+        except Government.DoesNotExist:
+            raise ValidationError(
+                "Government '%s' does not exist in sphere '%s'." %
+                (value, self.sphere))
+
+        return government
+
+
+class DepartmentInstanceLoader(ModelInstanceLoader):
+    """
+    Class to find a Department model instance from a row.
+    """
+
+    def get_instance(self, row):
+        """
+        Gets a Department instance by either a unique government-slug or
+        government-name combination, or if is_vote_primary is True, from a
+        unique government-vote_number combination.
+        """
+        name = self.resource.fields['name'].clean(row)
+        slug = slugify(name)
+        government = self.resource.fields['government'].clean(row)
+
+        instance = None
+
+        try:
+            instance = Department.objects.all().get(name=name, government=government)
```

Comment:
why `.all().get()` and not just `.get()`?

Also a couple more of these below

Reply:
Sorry, don't know why I did that. Fixing it now.